### PR TITLE
Add trimming tests for System.Private.Xml changes

### PR DESCRIPTION
**Pull request by layomia**
Contributes to https://github.com/dotnet/runtime/issues/37677 (Strip the ILLinkTrim.xml file from System.Private.Xml - #35547)

### CODE REVIEW

File: src/libraries/System.Data.Common/tests/TrimmingTests/CreateSqlXmlReader.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using System.Data.SqlTypes;
+using System.IO;
+
+class Program
+{
+    static int Main(string[] args)
+    {
+        MemoryStream ms = new MemoryStream();
+        var sqlXml = new SqlXml(ms);
+        var xmlReader = sqlXml.CreateReader();
```
Comment (eerhardt) [inline src/libraries/System.Data.Common/tests/TrimmingTests/CreateSqlXmlReader.cs:10]: We should verify the `xmlReader` is valid - at least that it isn't `null`.

File: src/libraries/System.Data.Common/tests/TrimmingTests/CreateSqlXmlReader.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System.Data.SqlTypes;
```
Comment (eerhardt) [inline src/libraries/System.Data.Common/tests/TrimmingTests/CreateSqlXmlReader.cs:1]: Have we been putting a copyright on these tests? We probably should - there is a copyright on all the tests in the repo.

**Review by eerhardt — APPROVED** [review]
Looks good. Just a couple minor comments.

File: src/libraries/System.Data.Common/tests/TrimmingTests/CreateSqlXmlReader.cs (hunk 2 above)
Comment (layomia) [reply src/libraries/System.Data.Common/tests/TrimmingTests/CreateSqlXmlReader.cs:5]: No 😬 - will start now.

**Pull request merged**